src/Domain/Entity/PurchaseEntity.cs

```diff
@@ -0,0 +1,207 @@
+                    ErrorMessages.PurchaseEntityInvalidPurchaseLineItem);
+            }
+        }
+    }
+
+    private void ValidateVatAmount(double? value)
+    {
+        if (!value.HasValue) return;
+        if (value < 0)
+        {
+            throw new DomainValidationException("VatAmount", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityVatAmountPositive);
+        }
+
+        // Check if value is rounded to two decimal places
+        double roundedValue = Math.Round(value.Value, 2);
+        if (Math.Abs(roundedValue - value.Value) >
+            0.0001) // Allowing a small tolerance for floating-point arithmetic
+        {
+            throw new DomainValidationException("VatAmount", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityNumberRoundedToTwoDecimalPlaces);
+        }
+    }
+    
+    private void ValidatePaidAmount(double? value)
+    {
+        if (!value.HasValue) return;
+        if (value < 0)
+        {
+            throw new DomainValidationException("PaidAmount", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityPaidAmountPositive);
+        }
+
+        // Check if value is rounded to two decimal places
+        double roundedValue = Math.Round(value.Value, 2);
+        if (Math.Abs(roundedValue - value.Value) >
+            0.0001) // Allowing a small tolerance for floating-point arithmetic
+        {
+            throw new DomainValidationException("PaidAmount", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityNumberRoundedToTwoDecimalPlaces);
+        }
+    }
+    
+    private void ValidateInvoiceNumber(int? value)
+    {
+        if (!value.HasValue) return;
+
+        if (value < 0)
+        {
+            throw new DomainValidationException("InvoiceNumber", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityInvoiceNumberPositive);
+        }
+    }
+    
+    private void ValidateTransportFee(double? value)
+    {
+        if (!value.HasValue) return;
+        if (value < 0)
+        {
+            throw new DomainValidationException("TransportFee", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityTransportFeePositive);
+        }
+
+        // Check if value is rounded to two decimal places
+        double roundedValue = Math.Round(value.Value, 2);
+        if (Math.Abs(roundedValue - value.Value) >
+            0.0001) // Allowing a small tolerance for floating-point arithmetic
+        {
+            throw new DomainValidationException("TransportFee", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityNumberRoundedToTwoDecimalPlaces);
+        }
+    }
+
+    private void ValidateDate(DateOnly value)
+    {
+        var dateString = value.ToString("yyyy-MM-dd");
+        if (!DateOnly.TryParse(dateString, out _))
+        {
+            throw new DomainValidationException("Date", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityDateFormatInvalid);
+        }
+
```

Comment:
Dont we already have a common error messange for invalid date ? Or do purchase expect different format ?